packages/api/src/command/medical/cq-directory/parse-cq-directory-entry.ts

```diff
@@ -79,21 +80,16 @@ function getUrls(contained: Contained): XCUrls | undefined {
   return urls;
 }
 
-function getCoordinates(address: Address[] | undefined): { lat: number; lon: number } | undefined {
-  if (!address) return;
-  const coordinates = address.flatMap(a => {
-    if (a.extension?.url === ORG_POSITION) {
-      const position = a.extension?.valueCodeableConcept?.coding?.value?.position;
-      if (!position) return [];
-      const lat = parseFloat(position.latitude.value);
-      const lon = parseFloat(position.longitude.value);
-      if (lat < -90 || lat > 90) return [];
-      if (lon < -180 || lon > 180) return [];
-      return { lat, lon };
-    }
-  })[0];
-
-  return coordinates;
+function getCoordinates(address: Address[]): Coordinates | undefined {
+  const orgPosition = address.find(a => a.extension?.url === ORG_POSITION);
+  const position = orgPosition?.extension?.valueCodeableConcept?.coding?.value?.position;
+  if (!position) return;
+  const lat = parseFloat(position.latitude.value);
+  const lon = parseFloat(position.longitude.value);
+  if (lat < -90 || lat > 90) return;
+  if (lon < -180 || lon > 180) return;
+  if (isNaN(lat) || isNaN(lon)) return;
```

Comment:
I think want to put the `isNan` check to happen before `if (lat < -90 || lat > 90) return;`
It doesnt matter in this case but is better for future proofing + readability I think. 

If lat or lon is NaN, the range checks will pass, and even though it will properly return on the isNan check, seems better to move it up